app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -75,7 +75,20 @@ def content_and_puppet_match?
       end
 
       def inherited_attributes_with_katello
-        self.content_facet.kickstart_repository_id ||= hostgroup.inherited_kickstart_repository_id if content_facet.present?
+        if hostgroup.present?
+          if content_facet.present?
+            self.content_facet.kickstart_repository_id ||= hostgroup.inherited_kickstart_repository_id
+          else
+            # facent may not be present when created via autoprovisioning
+            if hostgroup.inherited_content_view_id && hostgroup.inherited_lifecycle_environment_id
+              build_content_facet(
+                :kickstart_repository_id => hostgroup.inherited_kickstart_repository_id,
+                :content_source_id => hostgroup.inherited_content_source_id,
```

Comment:
getting an error because of this line:

`unknown attribute 'content_source_id' for Katello::Host::ContentFacet.`

content_source_id is actually on the host object.